### ISSUE TITLE
Help Getting Lazystream into xTeVE

### ISSUE THREAD

**Issue by House923** [tarkah/lazystream#65]
I've been told that the creator of lazystream has it setup where it works together with xteve, and then can be set up through Plex DVR.  I have been trying to do this for a while now, but can't seem to actually get any video.

I have figured out how to get Lazystream to generate an m3u and xml file, and can setup xteve to read both of those, and can even get Plex to see the guide, see what games are playing, etc.  But there's never any video content in the m3u's.

Just wondering if you can give a bit of a quick rundown on how you can actually get lazystream to export the working m3u?  I must be missing some step that makes the m3u an actual working file.

Thanks in advance!

**Comment by tarkah** [comment]
We've created a docker image that setups xTeVe with lazystream and all the necessary little tweaks needed to get it working properly here:

https://github.com/taylorbourne/xteve_lazystream

If you want to set it up manually, there are a few important steps needed to get things in a working state:

- xTeVe needs to be set to use `ffmpeg`. We couldn't get it to work with VLC as the player. Here are the relevant settings for xTeVe config file
```
  "buffer": "ffmpeg",
  "buffer.size.kb": 5120,
  "buffer.timeout": 500,
  "ffmpeg.options": "-hide_banner -loglevel error -i [URL] -c copy -f mpegts pipe:1",
  "ffmpeg.path": "/usr/bin/ffmpeg",
```
- The hosts file updates need to be made, but in addition to this, FFMPEG will fail because it won't be able to verify the SSL cert of the proxied host. The following script from our docker updates the hosts file, but also adds those insecure certs to the known hosts file so it wont fail. You may need `gnutls-utils` installed if you don't already to do this
```
# Update hosts file
export IP=$(getent ahostsv4 freegamez.ga | awk '{ print $1 }' | head -1)

echo $IP playback.svcs.mlb.com >> /etc/hosts
echo $IP mf.svc.nhl.com >> /etc/hosts
echo $IP mlb-ws-mf.media.mlb.com >> /etc/hosts

# Update certs
rm -f /home/abc/.gnutls/known_hosts
rm -f /root/.gnutls/known_hosts

printf 'y\n' | gnutls-cli --tofu playback.svcs.mlb.com:443
printf 'y\n' | gnutls-cli --tofu mf.svc.nhl.com:443
printf 'y\n' | gnutls-cli --tofu mlb-ws-mf.media.mlb.com:443

install -D /root/.gnutls/known_hosts /home/abc/.gnutls/known_hosts

rm -rf /root/.gnutls
```

This took a lot of trial and error to figure out, so I hope I'm not missing any steps. Hope this helps!

**Comment by seanz251** [comment]
Hey - Reopening this as I'm attempting to do the same, couple of questions. Have any of the certificates/host domains changed? 

Is there anything stopping me from running this via a box in the cloud? Lazystream generates m3u - feeds to Xteve - Linked to Plex DVR. Watch and access via remote PC via Plex? I ask as I know some other scripts clarify that you have to watch/view any generated links on a local ip.

**Comment by tarkah** [comment]
Have you checked out https://github.com/taylorbourne/xteve_lazystream?

Its a docker container set up to handle all of this. You should be able to run it and it should __just work__ ™️

**Comment by seanz251** [comment]
> Have you checked out https://github.com/taylorbourne/xteve_lazystream?
> 
> Its a docker container set up to handle all of this. You should be able to run it and it should **just work** ™️

My current setup doesn't allow for that at the moment unfortunately :(

I've followed all your instructions, but keep getting no streams either via Plex/XteVe. Or when downloading the generated m3u and trying to run it locally in VLC.

**Comment by tarkah** [comment]
Darn, that's unfortunate. It really is a pain to set up. That description I give in the second comment is the best outline of things that need to be done.

You can also inspect that repo / Dockerfile to see how things are done.

**Comment by seanz251** [comment]
Yeah one of those things. 

I need to try confirm that you don’t have to watch/open the stream via the local ip where the link was generated though? That would probably explain my issue if so. As everything else is setup accordingly.

Right now even just running the script in the cloud and then pasting the generated link on my local pc doesn’t work?

Is there a way of contacting you privately?

**Comment by tarkah** [comment]
If you have it working with xteve / Plex locally, there's nothing stopping you from using it remotely with Plex.

If you want just play the links locally on your PC, you can use the play subcommand of this program. You just need DNS updated for that to work.

**Comment by seanz251** [comment]
The only other issue I had was that  'mlb-ws-mf.media.mlb.com:443'... didn't resolve
I got the error: Cannot resolve mlb-ws-mf.media.mlb.com:443: Name or service not known

Does my host file look correctly formatted/laid out otherwise?

127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
**REMOVED FOR SECURITY PERSONAL IP**
65.9.83.37 playback.svcs.mlb.com
65.9.72.10 nhl.playback.edge.bamgrid.com

**Comment by tarkah** [comment]
Those IPs are not correct. You're also missing the entry for the `mlb-ws-mf` hostname you said isn't working.

Ping `freegamez.ga` and use that as the IP address for all 3 entries.

**Comment by seanz251** [comment]
export IP=$(getent ahostsv4 freegamez.ga | awk '{ print $1 }' | head -1)

excuse my lack of knowledge, but is this in layman’s terms just pinging the freegamez.ga for an IP? (As you mentioned earlier)
